### PR TITLE
DOC-10454 N1QL to SQL++ Rebrand for JavaScript UDFs

### PR DESCRIPTION
**Pull request by RayOffiah**
Added sqlpp attribute
Changes source block language definitions to sqlpp

### CODE REVIEW

File: modules/guides/pages/create-user-defined-function.adoc [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-:description: Creating a N1QL User-Defined Function to call your JavaScript function.
+:description: Creating an {sqlpp} User-Defined Function to call your JavaScript function.
 :page-edition: Enterprise Edition
 :page-toclevels: 2
-:keywords: N1QL, UDF, user-defined function
+:keywords: N1QL, UDF, user-defined function, SQL++
 [abstract]
 {description}
 
 == Introduction
```
Comment (simon-dew) [inline modules/guides/pages/create-user-defined-function.adoc]: This section is empty?

File: modules/guides/pages/create-user-defined-function.adoc [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 
 [#creating-the-n1ql-udf-function]
-== Creating the N1QL User-Defined Function
+== Creating the {sqlpp} User-Defined Function
 
-Before you can run the JavaScript function you have created (see xref:create-javascript-library.adoc[]), you will need to create a N1QL User-Defined Function to reference it. 
-You can create a N1QL User-Defined Function by using:
+Before you can run the JavaScript function you have created (see xref:create-javascript-library.adoc[]), you will need to create an {sqlpp} User-Defined Function to reference it.
+You can create an {sqlpp} User-Defined Function by using:
```
Comment (simon-dew) [inline modules/guides/pages/create-user-defined-function.adoc]: See comments on previous pull request. If we specify that "SQL++" is pronounced "sequel plus plus", we don't have to change the indefinite article.

Google ngram viewer gives [more matches for "a SQL" than "an SQL"](https://books.google.com/ngrams/graph?content=a+SQL%2Can+SQL&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Ca%20SQL%3B%2Cc0%3B.t1%3B%2Can%20SQL%3B%2Cc0), especially in recent years.

File: modules/guides/pages/create-user-defined-function.adoc [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
 
 * the xref:tools:udfs-ui.adoc[UDF UI] in the Query Workbench.
-* the xref:n1ql:n1ql-language-reference/createfunction.adoc[standard N1QL `CREATE FUNCTION` DDL], using the _External Functions_ option to reference the Javascript function.
+* the xref:n1ql:n1ql-language-reference/createfunction.adoc[standard {sqlpp} `CREATE FUNCTION` DDL], using the _External Functions_ option to reference the Javascript function.
 * the REST-API to execute a xref:n1ql:n1ql-language-reference/createfunction.adoc[`CREATE FUNCTION`] call.
 
-For more information on N1QL User Defined Functions in general, read xref:n1ql:n1ql-language-reference/userfun.adoc[]
+For more information on {sqlpp} User Defined Functions in general, read xref:n1ql:n1ql-language-reference/userfun.adoc[]
```
Comment (simon-dew) [inline modules/guides/pages/create-user-defined-function.adoc]: Full stop at end?

File: modules/guides/pages/create-user-defined-function.adoc [hunk 4]
```diff
@@ -60,18 +60,18 @@ From this list select the library containing your function.
 REST API::
 +
 --
-Run a `curl` command from the shell to add a N1QL User-Defined Function that calls your Javascript function.
+Run a `curl` command from the shell to add an {sqlpp} User-Defined Function that calls your Javascript function.
```
Comment (simon-dew) [inline modules/guides/pages/create-user-defined-function.adoc]: As above, I think we should leave the indefinite article unchanged; it's one less thing to worry about.

File: modules/guides/pages/create-user-defined-function.adoc [hunk 5]
```diff
@@ -80,6 +80,6 @@ include::example$javascript-udfs/create-scoped-n1ql-udf.n1ql[]
 --
 ====
 
-NOTE: The N1QL User-Defined Function will take the same scope as the JavaScript UDF it is referencing.
+NOTE: The {sqlpp} User-Defined Function will take the same scope as the JavaScript UDF it is referencing.
 
-Having created your N1QL User-Defined Function, the xref:call-user-defined-function.adoc[next step] is to use  N1QL statement to call the function.
+Having created your {sqlpp} User-Defined Function, the xref:call-user-defined-function.adoc[next step] is to use an  {sqlpp} statement to call the function.
```
Comment (simon-dew) [inline modules/guides/pages/create-user-defined-function.adoc]: As above, I think we should leave the indefinite article unchanged; it's one less thing to worry about.

File: modules/javascript-udfs/pages/calling-javascript-from-n1ql.adoc [hunk 6]
```diff
@@ -8,14 +8,14 @@
 
 == Introduction
 
-Before you can call a JavaScript function, you must first create a N1QL User-Defined Function it.
-The process to do this is explained in the  xref:guides:create-user-defined-function.adoc#creating-the-n1ql-udf-function[Creating the N1QL User-Defined Function] section of our xref:guides:javascript-udfs.adoc[] guide.
+Before you can call a JavaScript function, you must first create a {sqlpp} User-Defined Function it.
```
Comment (simon-dew) [inline modules/javascript-udfs/pages/calling-javascript-from-n1ql.adoc]: To what it? Seem to be some words missing at the end here

```suggestion
Before you can call a JavaScript function, you must first create a {sqlpp} User-Defined Function to call it.
```

File: modules/javascript-udfs/partials/libraries-and-scopes.adoc [hunk 7]
```diff
@@ -22,6 +22,7 @@ Furthermore, access restrictions can be applied to scopes, so that only certain
 
 include::partial$diagrams.adoc[tags=udf-scopes-diagram]
 
-The user does not call a JavaScript function (shown here as `getBusinessDays(startDate, endDate)`) directly; a *N1QL User-Defined Function* is defined to act a reference caller to the JavaScript function. 
+The user does not call a JavaScript function (shown here as `getBusinessDays(startDate, endDate)`) directly;
+an *{sqlpp} User-Defined Function* is defined to act a reference caller to the JavaScript function.
```
Comment (simon-dew) [inline modules/javascript-udfs/partials/libraries-and-scopes.adoc]: Suggest leaving indefinite article unchanged (as above) for ease of conversion and consistency:

```suggestion
a *{sqlpp} User-Defined Function* is defined to act a reference caller to the JavaScript function.
```

**Review by simon-dew — CHANGES_REQUESTED** [review]
Thanks @RayOffiah . Some comments / suggestions

**Review by simon-dew — APPROVED** [review]
LGTM 👍 (with comments)

File: modules/javascript-udfs/pages/calling-javascript-from-n1ql.adoc (hunk 6 above)
Comment (simon-dew) [reply modules/javascript-udfs/pages/calling-javascript-from-n1ql.adoc]: Don't think this has been fixed?

File: modules/guides/pages/create-user-defined-function.adoc (hunk 1 above)
Comment (simon-dew) [reply modules/guides/pages/create-user-defined-function.adoc]: Don't think this has been fixed? The **Introduction** section is still empty

Comment (RayOffiah) [reply modules/guides/pages/create-user-defined-function.adoc]: Mmm. Seems to be okay.

Comment (simon-dew) [reply modules/guides/pages/create-user-defined-function.adoc]: My bad 👍

**Review by simon-dew — APPROVED** [review]
LGTM 👍

**Pull request merged**